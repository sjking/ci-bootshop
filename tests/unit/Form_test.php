<?php namespace Generator;

include_once(dirname(dirname(__DIR__)) . '/Template/Form.php');
include_once(dirname(dirname(__DIR__)) . '/Model/FormModel.php');

$name = 'foobar';
$table = 'foobar_table';

$fields = array();
$fields[] = new FormFieldModel('id', 'input', 
	array('class' => 'form-control', 'id' => 'id-input'));
$fields[] = new FormFieldModel('name', 'input', 
	array('class' => 'form-control', 'id' => 'name-input'));
$fields[] = new FormFieldModel('status', 'dropdown',
	array('class' => 'form-control', 'id' => 'status-select'));
$label_params = array('class' => 'control-label');

$id = 'id';

$params = array('id' => 'foobar-form', 'class' => 'form-horizontal');

$form_model = new FormModel($name, $table, $fields, $id, $params, $label_params);

$form = new Form($form_model);


$output = $form->generate();

echo "output:\n" . $output . "\n";

?>