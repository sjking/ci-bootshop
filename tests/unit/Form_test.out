<form id="foobar-form" class="form-horizontal">
  <label for="id" class="control-label">id</label>
  <input name="id" value="<?php echo $foobar_row['id']; ?>" class="form-control" id="id-input">
  <label for="name" class="control-label">name</label>
  <input name="name" value="<?php echo $foobar_row['name']; ?>" class="form-control" id="name-input">
  <label for="status" class="control-label">status</label>
  <select name="status" class="form-control" id="status-select">
    <?php foreach($foobar_row['status'] as $name => $val) { ?>
      <option value="<?php echo $val; ?>" <?php echo $val == $foobar_row['status']_selected ? "selected" : null ?>><?php echo $name; ?></option>
    <?php } ?>
  </select>
</form>